src/Elastic.Apm.AspNetCore/ApmMiddleware.cs

```diff
@@ -88,6 +87,31 @@ public async Task InvokeAsync(HttpContext context)
 					Headers = responseHeaders
 				};
 
+				if (context.User?.Identity != null && context.User.Identity.IsAuthenticated && context.User.Identity != null
+					&& transaction.Context.User == null)
+				{
+					transaction.Context.User = new User { UserName = context.User.Identity.Name };
+
+					var nameId = GetClaimValue(ClaimTypes.NameIdentifier);

```

Comment:
This could be more optimal
```cs
					transaction.Context.User.Id = GetClaimWithFallbackValue(ClaimTypes.NameIdentifier, "sub");
					transaction.Context.User.Email = GetClaimWithFallbackValue(ClaimTypes.Email, "email");
				}

				string GetClaimWithFallbackValue(string claimType, string fallbackClaimType)
				{
					var idQuery = context.User.Claims.Where(n => n.Type == claimType || n.Type == fallbackClaimType);
					var idClaims = idQuery.ToList();
					return idClaims.Any() ? idClaims.First().Value : string.Empty;
				}

Reply:
Ok, let's do it this way. Pushed an update.